### PR TITLE
copied over janitor code for use in shakedown. 

### PR DESCRIPTION
**Pull request by kensipe**
it works for mesos versions greater than 0.28

need thoughts on error handling
it has been tested working against the confluence-kafka service.

### CODE REVIEW

**Comment by nickbp** [comment]
Worried that this copy will fall out of sync with the `janitor` image that users invoke. Any reason to not just invoke that?

**Comment by kensipe** [comment]
shakedown intends to be 100% python unless we can't.   There is also a strong desire to not be dependent on docker.     It sounds like both of these solutions will not be necessary after the next release, indicating a short term fear that is unlikely to see a change in this functionality and hence unlikely to fall out of sync.

**Comment by kensipe** [comment]
@mesosphere-ci retest this please

**Comment by kensipe** [comment]
@mesosphere-ci retest this please

**Comment by kensipe** [comment]
@sschneid the vmpooler keeps failing here... is it up?

**Comment by kensipe** [comment]
@mesosphere-ci retest this please

**Comment by sschneid** [comment]
@mesosphere-ci retest this please

File: shakedown/dcos/package.py [hunk 1]
```diff
@@ -257,6 +257,62 @@ def uninstall_package_and_wait(
...
+    delete_persistent_data(role, zk_node)
+
+    finish = time.time()
+
+    print('\n{}uninstall/delete done after pkg({}) + data({}) = total({})\n'.format(
+        shakedown.cli.helpers.fchr('>>'),
+        pretty_duration(data_start start),
+        pretty_duration(finish data_start),
+        pretty_duration(finish start)))
+
```
Comment (sschneid) [inline shakedown/dcos/package.py]: Syntax errors here.

**Comment by sschneid** [comment]
>   File "/jenkins/workspace/public-shakedown-pulls/channel/testing/master/label/py35/variant/open/shakedown/dcos/package.py", line 311
>     pretty_duration(data_start start),
>                                    ^
> SyntaxError: invalid syntax

**Comment by kensipe** [comment]
that's on me.. 1 sec

**Comment by kensipe** [comment]
@mesosphere-ci retest this please

**Comment by kensipe** [comment]
@mesosphere-ci retest this please

**Comment by kensipe** [comment]
@mesosphere-ci retest this please

**Comment by sschneid** [comment]
Failures unrelated to test.

**Pull request merged**